lib/record_store/provider/dynect.rb

```diff
@@ -34,7 +34,8 @@ def apply_changeset(changeset, stdout = $stdout)
 
       # returns an array of Record objects that match the records which exist in the provider
       def retrieve_current_records(zone:, stdout: $stdout)
-        session.get_all_records(zone).body.fetch('data').flat_map do |type, records|
+        session.get_all_records(zone).body.fetch('data').flat_map\
```

Comment:
I don't think this will be needed in the refactor if you rename the variable with a `_` prefix instead of ignoring the rubocop error here.